### PR TITLE
doc_cfg support and doc fix

### PR DESCRIPTION
**Pull request by zys864**
## Motivation
features tag on docs.rs support

cargo doc warning

## Solution
add doc_cfg support

doc fix

### CODE REVIEW

File: motore/Cargo.toml [hunk 1]
```diff
@@ -33,3 +33,7 @@ http = "0.2"
 
 [features]
 tower = ["dep:tower"]
+
+[package.metadata.docs.rs]
+all-features = true
+rustdoc-args = ["--cfg", "docsrs"]
```
Comment (PureWhiteWu) [inline motore/Cargo.toml:39]: Please add a newline to end of file.

**Review by PureWhiteWu — COMMENTED** [review]
Thank you very much for your contribution!
Just a little problem.

**Comment by Millione** [comment]
I think this file https://github.com/cloudwego/motore/blob/main/motore/src/service/tower_adapter.rs can also add the doc_cfg.

File: motore/Cargo.toml (hunk 1 above)
Comment (zys864) [reply motore/Cargo.toml]: already add it

Comment (zys864) [reply motore/Cargo.toml]: [/// This is similar to [`BoxService`](super::BoxService) except the resulting](/// This is similar to [`BoxService`](super::BoxService) except the resulting)

This line mention `BoxService`, but `BoxService` doesn't impl in super::BoxService.

**Review by PureWhiteWu — APPROVED** [review]
Thank you!
LGTM

**Pull request merged**